pkg/operator/status.go

```diff
@@ -23,6 +23,15 @@ const (
 	ReasonCheckAutoscaler   = "UnableToCheckAutoscalers"
 )
 
+// StatusReporterInterface is used to enable unit testing and does not
+// implement all methods of StatusReporter.
+type StatusReporterInterface interface {
+	CheckMachineAPI() (bool, error)
+	Fail(string, string) error
+	Progressing() error
+	Available(string, string) error
+}
```

Comment:
I see all of the listed methods to be used solely in `pkg/operator` package. Can we make all of them private?